src/components/AutoSuggestion/ExtendedCodeEditor.tsx

```diff
@@ -1,12 +1,19 @@
 import React from "react";
-import { UnControlled as UnControlledEditor } from "react-codemirror2";
 import { Classes as BlueprintClassNames } from "@blueprintjs/core";
-import { Editor as CodeMirrorEditor, EditorChange } from "codemirror";
-
-import "codemirror/addon/display/placeholder.js";
-import "codemirror/mode/sparql/sparql.js";
+import { indentWithTab } from "@codemirror/commands";
+import { EditorState } from "@codemirror/state";
+import { EditorView, lineNumbers } from "@codemirror/view";
+import { quietlight } from "@uiw/codemirror-theme-quietlight";
+import CodeMirror, { Extension, Rect, Statistics } from "@uiw/react-codemirror";
```

Comment:
Why don't we use our `CodeEditor` editor as foundation for `ExtendedCodeEditor` as specialized version?